pkg/runtime/values/helpers.go

```diff
@@ -328,24 +263,22 @@ func ToBoolean(input core.Value) core.Value {
 	}
 }
 
-func ToArray(input core.Value) core.Value {
+func ToArray(ctx context.Context, input core.Value) (core.Value, error) {
 	switch input.Type() {
 	case types.Boolean,
 		types.Int,
 		types.Float,
 		types.String,
 		types.DateTime:
 
-		return NewArrayWith(input)
-	case types.HTMLElement,
-		types.HTMLDocument:
-		val := input.(HTMLNode)
-		attrs := val.GetAttributes()
-
-		obj, ok := attrs.(*Object)
+		return NewArrayWith(input), nil
+	case types.Array:
+		return input.Copy(), nil
+	case types.Object:
+		obj, ok := input.(*Object)
```

Comment:
Valid point. But if a value has type of ``type.Object`` but isn't ``values.Object`` it means that someone messed up with types.
The reason why I created ``Type`` interface is that I wanted to avoid type assertion as much as I could because of its possible (in some cases) negative impact on performance.

Reply:
But ideally, we need to write benchmarks to see the impact. It seems in recent version of Go, it has been improved.

Reply:
I don't think this will be main performance problem, but OK :)